runtime/pkg/duckdbreplicator/io.go

```diff
@@ -0,0 +1,75 @@
+package duckdbreplicator
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+)
+
+// copyDir copies a directory from source to destination
+// It recursively copies all the contents of the source directory to the destination directory.
+// Files with the same name in the destination directory will be overwritten.
+func copyDir(dst, src string) error {
```

Comment:
As also mentioned in other comments, do we really need to be copying whole directories on local? Between local and the remote backup makes sense, but given that we never mutate files after they are initially created, it feels like we should be able to maintain just one copy of each file on local.